Sming/Arch/Esp8266/Components/spi_flash/flashmem.c

```diff
@@ -17,6 +17,7 @@
 
 #include "include/esp_spi_flash.h"
 #include "espinc/peri.h"
+#include <rboot-api.h>
```

Comment:
@mikee47 I like this PR, but what bothers me is the coupling between spi_flash and rBoot. At some point in time we can move to another bootloader or have a choice of bootloaders. I would prefer to have this code outside of `flashmem.c`. Every bootloader should have its own implementation. During linking time the linker can choose the right implementation of `flashmem_get_address`. 

Does that make sense to you?

Reply:
@slaff Yes, I agree, although it seems pretty unlikely that rBoot would get replaced for the ESP8266... Obviously the ESP32 is another matter.

What I would have preferred to do is query the hardware mapping register directly, but failed to find the relevant information; I included the `Cache_Read_Enable` document from Richard's blog for exactly this reason as so far it's the best research on the issue I've found so far.

For the moment I think co-operation with the bootloader is the safest way; I needed this fix for a deployed system so at some point I'll do some reverse-engineering on `Cache_Read_Enable` - unless someone else beats me to it :-) Then we can ditch the dependency and implement it 'properly' so it works independently of the bootloader.

Reply:
> What I would have preferred to do is query the hardware mapping register directly

Is that what you are looking for: excerpt from [Cache_Read_Enable.mw](https://github.com/pfalcon/esp8266-re-wiki-mirror/blob/master/Cache_Read_Enable.mw) says "Implementation writes to two [[Memory_Map#dport0_.283FF00000h.E2.80.93.29|dport0 registers]] to configure page mapping."

Reply:
Thanks, the address was already in the readme, just need a little clarification so have updated the doc and posted some code below.